core/test/element_array_view_test.cpp

```diff
@@ -113,265 +103,236 @@ auto range(const scipp::index end) {
+  EXPECT_TRUE(
+      equals(ElementArrayView(range(6).data(), 0, target,
+                              transposed_strides(dims, target.labels())),
+             {0, 3, 1, 4, 2, 5}));
 }
 
 TEST(ElementArrayViewTest, transpose_3d_yx) {
   Dimensions dims{{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}};
   Dimensions target({Dim::Y, Dim::X, Dim::Z}, {3, 2, 4});
-  EXPECT_TRUE(equals(ElementArrayView(range(24).data(), 0, target, dims),
-                     {0,  1,  2,  3,  12, 13, 14, 15, 4,  5,  6,  7,
-                      16, 17, 18, 19, 8,  9,  10, 11, 20, 21, 22, 23}));
+  EXPECT_TRUE(
+      equals(ElementArrayView(range(24).data(), 0, target,
+                              transposed_strides(dims, target.labels())),
+             {0,  1,  2,  3,  12, 13, 14, 15, 4,  5,  6,  7,
+              16, 17, 18, 19, 8,  9,  10, 11, 20, 21, 22, 23}));
 }
 
 TEST(ElementArrayViewTest, transpose_3d_zy) {
   Dimensions dims{{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}};
   Dimensions target({Dim::X, Dim::Z, Dim::Y}, {2, 4, 3});
-  EXPECT_TRUE(equals(ElementArrayView(range(24).data(), 0, target, dims),
-                     {0,  4,  8,  1,  5,  9,  2,  6,  10, 3,  7,  11,
-                      12, 16, 20, 13, 17, 21, 14, 18, 22, 15, 19, 23}));
+  EXPECT_TRUE(
+      equals(ElementArrayView(range(24).data(), 0, target,
+                              transposed_strides(dims, target.labels())),
+             {0,  4,  8,  1,  5,  9,  2,  6,  10, 3,  7,  11,
+              12, 16, 20, 13, 17, 21, 14, 18, 22, 15, 19, 23}));
 }
 
 TEST(ElementArrayViewTest, transpose_3d_zx) {
   Dimensions dims{{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}};
   Dimensions target({Dim::Z, Dim::Y, Dim::X}, {4, 3, 2});
-  EXPECT_TRUE(equals(ElementArrayView(range(24).data(), 0, target, dims),
-                     {0, 12, 4, 16, 8,  20, 1, 13, 5, 17, 9,  21,
-                      2, 14, 6, 18, 10, 22, 3, 15, 7, 19, 11, 23}));
+  EXPECT_TRUE(
+      equals(ElementArrayView(range(24).data(), 0, target,
+                              transposed_strides(dims, target.labels())),
+             {0, 12, 4, 16, 8,  20, 1, 13, 5, 17, 9,  21,
+              2, 14, 6, 18, 10, 22, 3, 15, 7, 19, 11, 23}));
 }
 
 TEST(ElementArrayViewTest, transpose_3d_zxy) {
   Dimensions dims{{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}};
   Dimensions target({Dim::Z, Dim::X, Dim::Y}, {4, 2, 3});
-  EXPECT_TRUE(equals(ElementArrayView(range(24).data(), 0, target, dims),
-                     {0, 4, 8,  12, 16, 20, 1, 5, 9,  13, 17, 21,
-                      2, 6, 10, 14, 18, 22, 3, 7, 11, 15, 19, 23}));
+  EXPECT_TRUE(
+      equals(ElementArrayView(range(24).data(), 0, target,
+                              transposed_strides(dims, target.labels())),
+             {0, 4, 8,  12, 16, 20, 1, 5, 9,  13, 17, 21,
+              2, 6, 10, 14, 18, 22, 3, 7, 11, 15, 19, 23}));
 }
 
 TEST(ElementArrayViewTest, collapse_inner) {
-  Dimensions dims{{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}};
+  // base dims = {{Dim::X, Dim::Y, Dim::Z}, {2, 3, 4}}
```

Comment:
commented code?

Reply:
This is meant to show what the underlying dimensions would be given the target dims and strides.

Reply:
But in all cases the comments seems to be a 1:1 copy of the line above? I don't understand which information it provides?